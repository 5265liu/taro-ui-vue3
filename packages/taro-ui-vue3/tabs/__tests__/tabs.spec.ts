import { mount } from '@vue/test-utils'
import AtTabs from '../src/index.vue'

const AXIOM = 'Rem is the best girl'

describe('AtTabs.vue', () => {
  test('render test', () => {
    const wrapper = mount(AtTabs, {
      slots: {
        default: AXIOM,
      },
    })
    expect(wrapper.text()).toEqual(AXIOM)
  })
})
