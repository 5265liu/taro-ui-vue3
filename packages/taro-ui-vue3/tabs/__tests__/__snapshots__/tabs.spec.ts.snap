// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtTabs Snap render AtTabs -- props animated 1`] = `
<view
  class="at-tabs--horizontal at-tabs--WEB at-tabs"
>
  <!-- with scroll view -->
  
  <!-- without scroll view -->
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    
    <view
      class="at-tabs__item at-tabs__item--active"
      id="tabtabs-AOTU20180"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页1
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20181"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页2
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20182"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页3
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    
  </view>
  
  <!-- tab body -->
  <view
    class="at-tabs__body"
    style="transform: translate3d(-0%, 0px, 0px); transition: unset;"
  >
    <view
      class="at-tabs__underline"
      style="width: 300%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs Snap render AtTabs -- props className 1`] = `
<view
  class="at-tabs--horizontal at-tabs--WEB at-tabs"
>
  <!-- with scroll view -->
  
  <!-- without scroll view -->
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    
    
  </view>
  
  <!-- tab body -->
  <view
    class="at-tabs__body"
    style="transform: translate3d(-0%, 0px, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="width: 0%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs Snap render AtTabs -- props current 1`] = `
<view
  class="at-tabs--horizontal at-tabs--WEB at-tabs"
>
  <!-- with scroll view -->
  
  <!-- without scroll view -->
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20180"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页1
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20181"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页2
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item at-tabs__item--active"
      id="tabtabs-AOTU20182"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页3
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    
  </view>
  
  <!-- tab body -->
  <view
    class="at-tabs__body"
    style="transform: translate3d(-200%, 0px, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="width: 300%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs Snap render AtTabs -- props customStyle 1`] = `
<view
  class="at-tabs--horizontal at-tabs--WEB at-tabs"
  customstyle="color:red;"
>
  <!-- with scroll view -->
  
  <!-- without scroll view -->
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    
    
  </view>
  
  <!-- tab body -->
  <view
    class="at-tabs__body"
    style="transform: translate3d(-0%, 0px, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="width: 0%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs Snap render AtTabs -- props height 1`] = `
<view
  class="at-tabs--vertical at-tabs--WEB at-tabs"
  style="height: 300px;"
>
  <!-- with scroll view -->
  
  <!-- without scroll view -->
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    
    <view
      class="at-tabs__item at-tabs__item--active"
      id="tabtabs-AOTU20180"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页1
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20181"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页2
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20182"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页3
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    
  </view>
  
  <!-- tab body -->
  <view
    class="at-tabs__body"
    style="height: 300px; transform: translate3d(0px, -0%, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="height: 300%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs Snap render AtTabs -- props swipeable 1`] = `
<view
  class="at-tabs--horizontal at-tabs--WEB at-tabs"
>
  <!-- with scroll view -->
  
  <!-- without scroll view -->
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    
    <view
      class="at-tabs__item at-tabs__item--active"
      id="tabtabs-AOTU20180"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页1
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20181"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页2
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20182"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页3
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    
  </view>
  
  <!-- tab body -->
  <view
    class="at-tabs__body"
    style="transform: translate3d(-0%, 0px, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="width: 300%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs Snap render AtTabs -- props tabDirection 1`] = `
<view
  class="at-tabs--vertical at-tabs--WEB at-tabs"
>
  <!-- with scroll view -->
  
  <!-- without scroll view -->
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    
    <view
      class="at-tabs__item at-tabs__item--active"
      id="tabtabs-AOTU20180"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页1
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20181"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页2
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20182"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页3
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    
  </view>
  
  <!-- tab body -->
  <view
    class="at-tabs__body"
    style="transform: translate3d(0px, -0%, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="height: 300%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs Snap render AtTabs -- props tabList 1`] = `
<view
  class="at-tabs--horizontal at-tabs--WEB at-tabs"
>
  <!-- with scroll view -->
  
  <!-- without scroll view -->
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    
    <view
      class="at-tabs__item at-tabs__item--active"
      id="tabtabs-AOTU20180"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页1
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20181"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页2
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20182"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页3
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    
  </view>
  
  <!-- tab body -->
  <view
    class="at-tabs__body"
    style="transform: translate3d(-0%, 0px, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="width: 300%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs Snap render initial AtTabs 1`] = `
<view
  class="at-tabs--horizontal at-tabs--WEB at-tabs"
>
  <!-- with scroll view -->
  
  <!-- without scroll view -->
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    
    
  </view>
  
  <!-- tab body -->
  <view
    class="at-tabs__body"
    style="transform: translate3d(-0%, 0px, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="width: 0%;"
    />
    
    
  </view>
</view>
`;
